pyslurm/slurm.pxd

```diff
@@ -452,229 +644,190 @@ cdef extern from 'slurm/slurm.h' nogil:
-        uint64_t *rpc_type_time
-        uint32_t rpc_user_size
-        uint32_t *rpc_user_id
-        uint32_t *rpc_user_cnt
-        uint64_t *rpc_user_time
-        uint32_t rpc_queue_type_count
-        uint32_t *rpc_queue_type_id
-        uint32_t *rpc_queue_count
-        uint32_t rpc_dump_count
-        uint32_t *rpc_dump_types
-        char **rpc_dump_hostlist
+    int slurm_hostlist_find(hostlist_t hl, char* hostname)
 
-    ctypedef stats_info_response_msg stats_info_response_msg_t
+    int slurm_hostlist_push(hostlist_t hl, char* hosts)
 
-    #
-    # Place holders for opaque data types
-    #
+    int slurm_hostlist_push_host(hostlist_t hl, char* host)
 
-    ctypedef struct xlist:
-        pass
+    ssize_t slurm_hostlist_ranged_string(hostlist_t hl, size_t n, char* buf)
 
-    ctypedef xlist *List
+    char* slurm_hostlist_ranged_string_malloc(hostlist_t hl)
 
-    ctypedef struct listIterator:
-        pass
+    char* slurm_hostlist_ranged_string_xmalloc(hostlist_t hl)
 
-    ctypedef listIterator *ListIterator
+    char* slurm_hostlist_shift(hostlist_t hl)
 
-    ctypedef void (*ListDelF) (void *x)
+    void slurm_hostlist_uniq(hostlist_t hl)
 
-    ctypedef int (*ListCmpF) (void *x, void *y)
+    ctypedef xlist* List
 
-    ctypedef int (*ListFindF) (void *x, void *key)
+    ctypedef listIterator* ListIterator
 
-    ctypedef int (*ListForF) (void *x, void *arg)
+    ctypedef void (*ListDelF)(void* x)
 
-    ctypedef struct job_resources:
-        pass
+    ctypedef int (*ListCmpF)(void* x, void* y)
 
-    ctypedef job_resources job_resources_t
+    ctypedef int (*ListFindF)(void* x, void* key)
 
-    ctypedef struct select_jobinfo:
-        pass
+    ctypedef int (*ListForF)(void* x, void* arg)
 
-    ctypedef select_jobinfo select_jobinfo_t
+    void* slurm_list_append(List l, void* x)
 
-    ctypedef struct select_nodeinfo:
-        pass
+    int slurm_list_count(List l)
 
-    ctypedef select_nodeinfo select_nodeinfo_t
+    List slurm_list_create(ListDelF f)
 
-    ctypedef struct jobacctinfo:
-        pass
+    void slurm_list_destroy(List l)
 
-    ctypedef jobacctinfo jobacctinfo_t
+    void* slurm_list_find(ListIterator i, ListFindF f, void* key)
+
+    int slurm_list_is_empty(List l)
+
+    ListIterator slurm_list_iterator_create(List l)
 
-    ctypedef struct hostlist:
-        pass
+    void slurm_list_iterator_reset(ListIterator i)
 
-    ctypedef hostlist *hostlist_t
+    void slurm_list_iterator_destroy(ListIterator i)
 
-    ctypedef struct dynamic_plugin_data:
-        void *data
+    void* slurm_list_next(ListIterator i)
+
+    void slurm_list_sort(List l, ListCmpF f)
+
+    void* slurm_list_pop(List l)
+
+    ctypedef int64_t bitstr_t
+
+    ctypedef bitstr_t bitoff_t
+
+    cdef struct dynamic_plugin_data:
+        void* data
         uint32_t plugin_id
 
     ctypedef dynamic_plugin_data dynamic_plugin_data_t
 
-    ctypedef struct acct_gather_energy:
+    cdef struct acct_gather_energy:
```

Comment:
why `cdef` and not `ctypedef` for declaring structs?

Reply:
Same reason as above, it is generated more or less automatically on the base of slurm.j2.

Reply:
So, the autopxd tool decided this?

Reply:
Yes, I think so.